transport/redis/inbound.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package redis
+
+import "go.uber.org/yarpc/transport"
+
+type inbound struct {
+	server  QueueServer
+	handler *handler
+}
+
+//NewInbound creates a redis transport.Inbound
+func NewInbound(server QueueServer) transport.Inbound {
+	return &inbound{server: server}
+}
+
+func (i *inbound) Start(service transport.ServiceDetail, deps transport.Deps) error {
+	err := i.server.Start()
+	if err != nil {
+		return err
+	}
+
+	i.handler = newHandler(service, i.server, deps)
```

Comment:
Optional: The reason `handler` exists in HTTP and TChannel is because they
already have a concept of "this interface should receive the HTTP/TChannel
request".  Since you're hand-rolling this, you don't need to define that
`handler` object. As it stands right now, you're not getting any benefit by
keeping `handler` and `inbound` separate; no extra testability or readability.